src/Bicep.Core/Emit/CompileTimeImports/ArmTemplateHelpers.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Deployments.Core.Definitions.Schema;
+using Bicep.Core.Workspaces;
+
+namespace Bicep.Core.Emit.CompileTimeImports;
+
+internal static class ArmTemplateHelpers
+{
+    private const string ArmTypeRefPrefix = "#/definitions/";
+    internal static SchemaValidationContext ContextFor(ArmTemplateFile templateFile)
+    {
+        if (templateFile.Template is not {} template)
+        {
+            throw new InvalidOperationException($"Source template of {templateFile.FileUri} is not valid");
+        }
+
+        return SchemaValidationContext.ForTemplate(template);
+    }
+
+    internal static ITemplateSchemaNode DerefArmType(SchemaValidationContext context, string typePointer)
+    {
+        // TODO make LocalSchemaRefResolver in Azure.Deployments.Templates public
+        if (!typePointer.StartsWith(ArmTypeRefPrefix) ||
+            typePointer.Substring(ArmTypeRefPrefix.Length).Contains('/') ||
+            !context.Definitions.TryGetValue(typePointer.Substring(ArmTypeRefPrefix.Length), out var typeDefinition))
+        {
```

Comment:
Can use range operator:
```suggestion
            typePointer[ArmTypeRefPrefix.Length..].Contains('/') ||
            !context.Definitions.TryGetValue(typePointer[ArmTypeRefPrefix.Length..], out var typeDefinition))
        {
```